### ISSUE TITLE
dev: tests should not fail fast in CI to allow devs to know exactly which tests break

### ISSUE THREAD

**Issue by Eikix** [kkrt-labs/kakarot-rpc#1030]
**Describe the enhancement request**

Tests should not fail fast in CI to allow devs to know exactly which tests break.
Currently in CI, if there is an error in the code that break tests, the test suite will stop at the first error. We would like to have more information when iterating on a PR.

**Describe Preferred Solution**

Make sure tests continue till the end.
Try adding the `--no-fail-fast` flag to the cargo test command.

**Comment by greged93** [comment]
I don't think we should encourage devs to use CI to debug. CI is the last barrier to check before merging, it shouldn't be used as a debug tool. Devs should be able to run all the tests locally with `make test` and then only pushing their changes.

**Comment by Eikix** [comment]
Closing for now
